src/Carousel.js

```diff
@@ -0,0 +1,69 @@
+import React, {useState} from 'react';
+import styled from 'styled-components';
+import ButtonSlider from './ButtonSlider';
+
+const CarouselComponent = ({className, data}) => {
+    const [xPos, setXpos] = useState(0);
+
+    const moveCarousel = (direction) => {
```

Comment:
Have you considered transforming this to a hook?

Reply:
I haven't learned about it yet haha. I'm consuming the material atm.

Reply:
Done.